presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeHiveExternalTableQueries.java

```diff
@@ -0,0 +1,119 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.nativeworker;
+
+import com.facebook.presto.testing.MaterializedResult;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.DistributedQueryRunner;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createCustomer;
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createLineitem;
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createNationWithFormat;
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createOrders;
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createPart;
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createPartSupp;
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createRegion;
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createSupplier;
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.dropTableIfExists;
+import static com.facebook.presto.nativeworker.PrestoNativeQueryRunnerUtils.createExternalTable;
+import static com.facebook.presto.nativeworker.SymlinkManifestGeneratorUtils.cleanupSymlinkData;
+import static com.facebook.presto.nativeworker.SymlinkManifestGeneratorUtils.getTpchTableColumns;
+import static java.lang.String.format;
+import static org.testng.Assert.assertEquals;
+
+public abstract class AbstractTestNativeHiveExternalTableQueries
+        extends AbstractTestNativeTpchQueries
+{
+    private static final String HIVE = "hive";
+    private static final String TPCH = "tpch";
+    private static final String SYMLINK_FOLDER = "symlink_tables_manifests";
+    private static final String HIVE_DATA = "hive_data";
+
+    @Override
+    protected void createTables()
+    {
+        QueryRunner javaQueryRunner = (QueryRunner) getExpectedQueryRunner();
+        createOrders(javaQueryRunner);
+        createLineitem(javaQueryRunner);
+        createNationWithFormat(javaQueryRunner, "PARQUET");
+        createCustomer(javaQueryRunner);
+        createPart(javaQueryRunner);
+        createPartSupp(javaQueryRunner);
+        createRegion(javaQueryRunner);
+        createSupplier(javaQueryRunner);
+
+        createExternalTable(javaQueryRunner, TPCH, "orders", getTpchTableColumns("orders"));
```

Comment:
Maybe create a list of the tpch table names and iterate over it for the create and drop table calls.